### PR TITLE
Enabling Windows + containerd + csi-proxy azure-disk/azure-file e2e jobs

### PR DESCRIPTION
**Pull request by marosset**
Adding e2e jobs that validation azure-disk/azure-file CSI plugins on clusters configured with:
- Windows
- Containerd
- CSI-proxy

### CODE REVIEW

**Comment by marosset** [comment]
@chewong, should we disable the period jobs for the in-tree azure-file/azure-disk driver (or run then less frequently)?

**Comment by marosset** [comment]
related to https://github.com/kubernetes-csi/csi-proxy/issues/55

File: config/jobs/kubernetes-sigs/sig-windows/sig-windows-config.yaml [hunk 1]
```diff
@@ -589,6 +589,69 @@ periodics:
...
+      - --aksengine-template-url=https://github.com/kubernetes-sigs/windows-testing/blob/master/job-templates/kubernetes_containerd_csi_proxy.json
+      - --aksengine-win-binaries
+      - --aksengine-deploy-custom-k8s
+      - --aksengine-agentpoolcount=2
+      # Specific test args
+      - --test-azure-disk-csi-driver
+      securityContext:
+        privileged: true
+      env:
+      - name: AZURE_STORAGE_DRIVER
```
Comment (chewong) [inline config/jobs/kubernetes-sigs/sig-windows/sig-windows-config.yaml:646]: We should be testing `disk.csi.azure.com` right? If so, we don't need to include the `AZURE_STORAGE_DRIVER` environment variable.

**Review by chewong — APPROVED** [review]
/lgtm

**Comment by andyzhangx** [comment]
azure file/disk on Windows containerd actually does not work now

**Pull request merged**